crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -0,0 +1,764 @@
+// Copyright (c) 2024 - Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::cluster_controller::scheduler::ObservedClusterState;
```

Comment:
If you move this to a shared branch, I can help with trivial updates like fixing up imports order :-)

Reply:
I'll move it to a separate module and then we can iterate on the improvements together.